applications/serial_lte_modem/src/twi/slm_at_twi.c

```diff
@@ -58,8 +60,8 @@ static int do_twi_write(uint16_t index, uint16_t dev_addr,
 {
 	int ret = -EINVAL;
 
-	if (!slm_twi_dev[index]) {
-		LOG_ERR("TWI device is not opened");
+	if (index >= ARRAY_SIZE(slm_twi_dev)) {
```

Comment:
Similar problem here (and below). `index` is the TWIM hardware instance index, so `if (!slm_twi_dev[index]) {` should stay and `slm_twi_dev` should be initialized with designated initializers.
See:
https://github.com/nrfconnect/sdk-nrf/blob/84e83748cda5ebd41b85d590225b5f6741d3e614/applications/serial_lte_modem/doc/TWI_AT_commands.rst#L75-L82

Reply:
I see, thanks for spotting this!